captum/optim/_core/objectives.py

```diff
@@ -196,6 +201,45 @@ def loss_function(targets_to_values: ModuleOutputMapping):
     return loss_function
 
 
+def deepdream(target: nn.Module) -> LossFunction:
+    def loss_function(targets_to_values: ModuleOutputMapping):
+        activations = targets_to_values[target]
+        return activations ** 2
```

Comment:
I squared the activation as that's how Lucid does it, and I've reached out for clarification as to why they do it. I also performed a quick test to compare with and without squaring the activtions.

Without squaring the activations:

![output_no_pow](https://user-images.githubusercontent.com/10626398/98273301-568eab00-1f4f-11eb-8ba6-6cc6d98a33b0.png)

With squaring the activations:

![output_pow](https://user-images.githubusercontent.com/10626398/98273390-70c88900-1f4f-11eb-81b8-c070f3b9ce44.png)

Reply:
The shape of the activations shouldn't matter for this objective as it's just maximizing the activations at the target layer.

Reply:
I changed the power value to be a user specified value for now.

Reply:
I think for the moment we can leave it as a user controllable value for power, as I'm sure there's a good reason Ludwig and his team did it that way. But I'll continue looking into it.

Reply:
@greentfrapp just helped figure out what I think is probably the actual reason. Squaring the activations removes negative values.

Reply:
@NarineK Okay, so I spoke with Chris about it and if the square is removed, it'd be doing a direciton visualization of [1,1,1,....] instead of DeepDream. 

> DeepDream tries to increase the values of neurons proportional to the amount they are presently active which is equivalent to maximizing the sum of the squares.

Reply:
Thank you for clarifying this, @ProGamerGov! Perhaps, we can add that explanation in the code so that we don't forget it.